test/libsolidity/semanticTests/payable/no_nonpayable_circumvention_by_modifier.sol

```diff
@@ -0,0 +1,17 @@
+contract C {
+    modifier tryCircumvent {
+        if (false) _; // avoid the function, we should still not accept ether
+    }
+    function f() tryCircumvent public returns (uint) {
+        return msgvalue();
+    }
+    function msgvalue() internal returns (uint) {
+        return msg.value;
+    }
+    function balance() external returns (uint) {
+        return address(this).balance;
+    }
+}
+// ----
+// f(), 27 wei -> FAILURE
```

Comment:
Looks like the docs do not say what happens when a modifier prevents function from returning a value and does not return any on its own. Actually, I'm surprised that it's even allowed. We should document that.

Reply:
The return variables are initialised to their default value (so 0 for numbers), unless they are explicitly set by something or there's an explicit return statement which overrides those variables.

There is an issue arguing this should not be the case.

Reply:
But this does not return 0. It reverts.

Or is the revert due to something else? I see the function only accessing `msg.value` so I assumed the revert is due to the modifier not returning.

Reply:
I think the goal here is that `msg.value` is invalid in nonpayable internal? Honestly I'm not sure.